c7n/resources/licensemanager.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from c7n.actions import (ActionRegistry, BaseAction)
+from c7n.manager import resources
+from c7n.query import QueryResourceManager
+from c7n.utils import local_session, type_schema
+from c7n.tags import RemoveTag, Tag
+
+actions = ActionRegistry('license-configuration.actions')
+
+
+@resources.register('license-configuration')
+class LicenseConfiguration(QueryResourceManager):
+
+    class resource_type(object):
+        service = 'license-manager'
+        enum_spec = ('list_license_configurations', 'LicenseConfigurations', None)
+        detail_spec = (
+            'get_license_configuration', 'LicenseConfigurationArn',
+            'LicenseConfigurationArn', None)
+        id = 'LicenseConfigurationArn'
+        name = 'Name'
+        date = None
+        dimension = None
+        filter_name = None
+
+    permissions = ('license-manager:ListTagsForResource',)
+
+    def augment(self, resources):
+        client = local_session(self.session_factory).client('license-manager')
+
+        def _augment(r):
+            # List tags for the license_configuration & set as attribute
+            tags = self.retry(client.list_tags_for_resource,
+                ResourceArn=r['LicenseConfigurationArn'])['Tags']
+            r['Tags'] = tags
+            return r
+
+        # Describe license_configuration & then list tags
+        resources = super(LicenseConfiguration, self).augment(resources)
+        with self.executor_factory(max_workers=1) as w:
```

Comment:
no need for the executor factory given resource cardinality or using max_workers=1 just do a for loop with _augment